docs:  improvements (#283)
nuxt/image: docs/content/en/2.setup.md

See [module options](/api/options).

## TypeScript

If you're using Typescript, add the types to your "types" array in `tsconfig.json` after the `@nuxt/types` (Nuxt 2.9.0+) or `@nuxt/vue-app` entry.

```json [tsconfig.json]
{
  "compilerOptions": {
    "types": ["@nuxt/types", "@nuxt/image"]
  }
}
```

## Troubleshooting

> To transform your images, Nuxt Image includes `lib sharp` and `vips` as dependencies. If errors occur on installation, you might be in these cases :

Reviewer: ```suggestion
To transform your images, Nuxt Image includes `sharp` as dependencies. If a node-gyp error occur on installation:
```

> - Your architecture is not supported by the pre-built binaries (e.g Apple M1). This means that NPM has to build them from source. On Apple systems, this requires XCode.

Reviewer: ```suggestion
- Your architecture is not supported by the pre-built binaries (e.g Apple M1). This means that npm has to build them from source. On Apple systems, this requires python and XCode, on Linux syatems python and gcc (`build-essentials`) and on windows, you can check installing requirements with node installer`
```

- Your lock file (`package-lock.json` or `yarn.lock`) references older versions of the libraries. Try to re-create your lock file by doing so:
  <code-group>
    <code-block label="Yarn" active>

  ```bash
  rm yarn.lock && yarn add
  ```

    </code-block>
    <code-block label="NPM">

  ```bash
  rm package-lock.json && npm install
  ```

    </code-block>
  </code-group>

> - Your project contains a previous version of Nuxt Image. If so, try to upgrade :

Reviewer: Ideally this should be before lock recreation suggest (first does second)

Reviewer: Node version could be nice to mentioned (14)

See [module options](/api/options).

## TypeScript

If you're using Typescript, add the types to your "types" array in `tsconfig.json` after the `@nuxt/types` (Nuxt 2.9.0+) or `@nuxt/vue-app` entry.

```json [tsconfig.json]
{
  "compilerOptions": {
    "types": ["@nuxt/types", "@nuxt/image"]
  }
}
```

## Troubleshooting

<alert type="info">
  We recommend using Node LTS version or 12, 14 or 16
</alert>

To transform your images, Nuxt Image includes `sharp` as a dependency. If a node-gyp error occurs on installation:

> - Your architecture is not supported by the pre-built binaries (e.g Apple M1). This means that npm has to build them from source. On Apple systems, this requires python and XCode, on Linux systems python and gcc (`build-essentials`) and on windows, you can check installing requirements with node installer`

Reviewer 2: ```suggestion
- Your architecture is not supported by the pre-built binaries (for example, this sometimes occurs on Apple M1). This means that npm has to build them from source. On Apple systems, this will require python and XCode, on Linux systems, python and gcc (`build-essentials`) and on Windows, you can check your installation requirements with the node installer.
```

Not sure what that last sentence means. cc: @pi0?

> - Your architecture is not supported by the pre-built binaries (for example, this sometimes occurs on Apple M1). This means that npm has to build them from source. On Apple systems, this will require python and XCode, on Linux systems, python and gcc (`build-essentials`) and on Windows, you can check your installation requirements with the node installer.

Reviewer: Todo for myself: windows and linux need improvement.

> - Your project contains a previous version of Nuxt Image. If so, try to upgrade :

Reviewer 2: ```suggestion
- Your project contains a previous version of Nuxt Image. If so, try to upgrade:
```

  <code-group>
    <code-block label="Yarn" active>

  ```bash
  yarn upgrade @nuxt/image
  ```

    </code-block>
    <code-block label="NPM">

  ```bash
  npm up @nuxt/image
  ```

    </code-block>
  </code-group>

> - Your lock file (`package-lock.json` or `yarn.lock`) references older versions of the libraries. Try to re-create your lock file by doing so:

Reviewer 2: ```suggestion
- Your lock file (`package-lock.json` or `yarn.lock`) references older versions of the libraries. Try to re-create your lock file:
```

  <code-group>
    <code-block label="Yarn" active>

  ```bash
  rm yarn.lock && yarn add
  ```

    </code-block>
    <code-block label="NPM">

  ```bash
  rm package-lock.json && npm install
  ```

    </code-block>
  </code-group>

> - If none of the above worked, please fill an issue here : [https://github.com/nuxt/image/issues](https://github.com/nuxt/image/issues). Please include error trace, OS, Node version and the package manager used for installing.

Reviewer 2: ```suggestion
- If none of the above worked, please [create an issue](https://github.com/nuxt/image/issues). Please include error trace, OS, Node version and the package manager used for installing.
```
